tools/ml-audio-content-moderation/terraform/main.tf

```diff
@@ -0,0 +1,253 @@
+# Copyright 2019 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+provider "google" {
+  project = "${var.projectid}"
+}
+
+# Enable APIs; Must be individual resources or else it will disable all other APIs for the project.
+resource "google_project_service" "billingapi" {
+  service = "cloudbilling.googleapis.com"
+}
+
+resource "google_project_service" "schedulerapi" {
+  service = "cloudscheduler.googleapis.com"
+}
+
+resource "google_project_service" "pubsubapi" {
+  service = "pubsub.googleapis.com"
+}
+
+resource "google_project_service" "cfapi" {
+  service = "cloudfunctions.googleapis.com"
+}
+
+resource "google_project_service" "bqapi" {
+  service = "bigquery-json.googleapis.com"
+}
+
+resource "google_project_service" "gaeapi" {
+  service = "appengine.googleapis.com"
+}
+
+resource "google_project_service" "nlpapi" {
+  service = "language.googleapis.com"
+}
+
+resource "google_project_service" "speechapi" {
+  service = "speech.googleapis.com"
+}
+
+resource "google_project_service" "commentanalyzerapi" {
+  service = "commentanalyzer.googleapis.com"
+}
+
+resource "random_uuid" "uuid" {}
+
+
+# Create GCS Buckets
+# Create GCS Bucket to hold audio files on upload before they are processed by pipeline.
+resource "google_storage_bucket" "staging_audio_bucket" {
+  provider = "google"
+  name = "staging-audio-files-${substr(lower(random_uuid.uuid.result), 0, 20)}"
```

Comment:
Instead of using random_uuid and taking a substring, just use https://www.terraform.io/docs/providers/random/r/id.html or https://www.terraform.io/docs/providers/random/r/pet.html

Reply:
Used random id & hex for 10 bytes, thanks!